testapi.pm

```diff
@@ -704,6 +705,85 @@ sub console($) {
     }
 }
 
+=head2 parse_junit_log
+
+parse_junit_log("report.xml")
+
+Parse jUnit log file and creates extra test result from it. The file must
+be first uploaded by upload_logs().
+
+upload_logs("/tmp/report.xml");
+parse_junit_log("report.xml");
+
+=cut
+
+
+sub parse_junit_log {
+    my ($file) = @_;
+
+    die "'/' is not allowed in file name" if $file =~ /\//;
+
+    open my $fd, "ulogs/$file" or die "Couldn't open file 'ulogs/$file': $!";
+    my $xml = join("", <$fd>);
+    close $fd;
+
+    my $dom = Mojo::DOM->new($xml);
+
+    my @tests;
+
+    for my $ts ($dom->find('testsuite')->each) {
+        my $ts_category = $ts->{package};
+        $ts_category =~ s/[^A-Za-z0-9._-]/_/g;    # the name is used as part of url so we must strip special characters
+        my $ts_name = $ts_category;
+        $ts_category =~ s/\..*$//;
+        $ts_name =~ s/^[^.]*\.//;
+        $ts_name =~ s/\./_/;
+
+        push @tests,
+          {
+            "flags"    => {"important" => 1},
+            "category" => $ts_category,
+            "name"     => $ts_name,
+            "script"   => $autotest::current_test->{script},
+          };
+
+        my $ts_result = 'ok';
+        $ts_result = 'fail' if $ts->{failures} || $ts->{errors};
+
+        my $result = {
+            "result"  => $ts_result,
+            "details" => [],
+            "dents"   => 0,
+        };
+
+        my $i = 1;
```

Comment:
It's not obvious what $i is - please pick a better name. It only becomes clear that it's a counter 28 lines later